src/AzureExtension/DevBox/Helpers/WingetConfigWrapper.cs

```diff
@@ -166,10 +185,23 @@ private void SetStateForCustomizationTask(TaskJSONToCSClasses.BaseClass response
             case "Running":
                 for (var i = 0; i < _units.Count; i++)
                 {
-                    var task = _units[i];
-                    var unitState = DevBoxOperationHelper.JSONStatusToUnitStatus(response.Tasks[i].Status);
-                    if (_oldUnitState[i] != unitState)
+                    var responseStatus = response.Tasks[i].Status;
+                    var unitState = DevBoxOperationHelper.JSONStatusToUnitStatus(responseStatus);
+
+                    // If the status is waiting for a user session, there is no need to check for other
+                    // individual task statuses. We add a wait since Winget takes time to start applying
+                    // the configuration and we don't want to show the same message immediately after.
+                    if (responseStatus == "WaitingForUserSession")
                     {
+                        ApplyConfigurationActionRequiredEventArgs eventArgs = new(new WaitingForUserAdaptiveCardSession(_resumeEvent));
```

Comment:
Why not incapsulate _resumeEvent into WaitingForUserAdaptiveCardSession? It's an implementation detail for WaitingForUserAdaptiveCardSession how to wait for completion and what result to return upon it.

Reply:
Could you elaborate on this? What should I change it to?

Reply:
Something like:
```
var adaptiveCardSession = new WaitingForUserAdaptiveCardSession();
ApplyConfigurationActionRequiredEventArgs eventArgs = new(adaptiveCardSession);
...
adaptiveCardSession.WaitForUserResponse(); // Can have a response value if needed
```

In WaitingForUserAdaptiveCardSession define _resumeEvent like it's here and 
```
public void WaitForUserResponse()
{
    WaitHandle.WaitAny(new[] { _resumeEvent });
}
```

Functionally it's the same, but less coupling between the caller and WaitingForUserAdaptiveCardSession. And if you'll change WaitingForUserAdaptiveCardSession to have more actions, like a "Cancel" button, WaitForUserResponse can return what action was chosen since it has access to all of the WaitingForUserAdaptiveCardSession implementation..